lib/locomotive/steam/middlewares/auth.rb

```diff
@@ -118,20 +118,23 @@ def append_message(message)
         liquid_assigns["auth_#{message}"] = "auth_#{message}"
       end
 
-      class AuthOptions
+      def auth_service
+        provider = site.metafields[:auth_provider]
+        if provider
+          @auth_service ||= services.send("auth_#{provider[:type]}")
```

Comment:
it's better to test if `provider[:type]` is among a list of authorized services before calling `.send(...)`.